### ISSUE TITLE
TransformedTransitionKernel can't use loglikelihood function with multiple arguments

### ISSUE THREAD

**Issue by rzu512** [tensorflow/probability#442]
The `TransformedTransitionKernel` can't work with log-likelihood function with multiple arguments.

If I don't use `TransformedTransitionKernel`, I guess the Hamiltonian Monte Carlo would use the same mass for all parameters. In a complex model, the gradient of the log-likelihood with respect to some parameters can be much steeper than the gradient w.r.t other parameters. That means some parameters are much less sampled than other parameters. Some parameters might get stuck near one value.

If I convert the signature `log_likelihood_fn(x, y)` to `log_likelihood_fn(vars)` where `vars = x, y`, I get this error: `Tensor objects are only iterable when eager execution is enabled.`

```python
import tensorflow as tf
import numpy as np
import tensorflow_probability as tfp

tfb = tfp.bijectors

"""
# this doesn't work
def log_likelihood_fn(vars):
    x, y = vars
    return tf.reduce_sum(x ** 2) + tf.reduce_sum(y ** 2)


def make_cur_state():
    return [np.ones((2, 3)), np.ones((4, 5, 6))]
"""

# this also doesn't work
def log_likelihood_fn(x, y):
    return tf.reduce_sum(x ** 2) + tf.reduce_sum(y ** 2)


def make_cur_state():
    return np.ones((2, 3)), np.ones((4, 5, 6))

def main():
    step_size = tf.get_variable(
        name='step_size',
        initializer=0.05,
        use_resource=True,  # For TFE compatibility.
        trainable=False)

    num_results = int(10)
    num_burnin_steps = int(10)

    cur_state = make_cur_state()

    step_size_update_fn = tfp.mcmc.make_simple_step_size_update_policy(
        num_adaptation_steps=int(num_burnin_steps * 0.8))

    hmc = tfp.mcmc.HamiltonianMonteCarlo(
        target_log_prob_fn=log_likelihood_fn,
        num_leapfrog_steps=3,
        step_size=step_size,
        step_size_update_fn=step_size_update_fn)

    bij_temp = tfb.real_nvp_default_template(hidden_layers=[512, 512, 512])
    bijector = tfb.RealNVP(num_masked=2, shift_and_log_scale_fn=bij_temp)
    realnvp_hmc = tfp.mcmc.TransformedTransitionKernel(
        inner_kernel=hmc, bijector=bijector)

    samples, kernel_results = tfp.mcmc.sample_chain(
        num_results=num_results,
        num_burnin_steps=num_burnin_steps,
        current_state=cur_state,
        kernel=realnvp_hmc)

    init_op = tf.global_variables_initializer()

    with tf.Session() as sess:
        sess.graph.finalize()
        init_op.run()
        samples_, kernel_results_ = sess.run([samples, kernel_results])


main()
```

**Comment by srvasude** [comment]
Hi,
  This should be fixed now. In particular the major change that needs to be made is TransformedTransitionKernel now needs to take a structure of bijectors (i.e. your state is list of shape [2], TTK needs a list of 2 bijectors, one for each part of the state).

Closing this as this should work now, but feel free to reopen if there any further issues.
